src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java

```diff
@@ -155,7 +156,19 @@ public HyperwalletUser updateUser(HyperwalletUser user) {
      * @return HyperwalletList of HyperwalletUser
      */
     public HyperwalletList<HyperwalletUser> listUsers(HyperwalletPaginationOptions options) {
+        return listUsers(options, new HashMap<String, String>());
+    }
+
+    /**
+     * List Users
+     *
+     * @param options List filter option
+     * @param params Custom filter options
+     * @return HyperwalletList of HyperwalletUser
+     */
+    public HyperwalletList<HyperwalletUser> listUsers(HyperwalletPaginationOptions options, Map<String, String> params) {
```

Comment:
Can you provide to me the use case so I can better understand why a generic params map is meant for?

Reply:
Sure. According to the documentation each "list endpoint" supports different query params to filter the result. I.e. the list users endpoint supports the email parameter while the list bank accounts endpoint supports the type parameter.

We could have well defined query classes for each endpoint only exposing the supported params. If it'd be up to me only I would go with the map implementation since it allows developers to use parameters that might have been introduced to the api but are not yet included in the the query classes.

Reply:
Following the design we have today, I'd prefer creating a **HyperwalletUserListOptions** that **extends HyperwalletPaginationOptions**.

I get when you say that this is way more flexible, but we want to keep our SDK self contained without requiring the developer to go check our documentation to understand what is available to be used and what isn't. That also helps us unit test it.

Reply:
Fine for me. I'll pick it up.

Reply:
Please check again. I'm not super happy with the current implementation but tried to prevent unnecessary changes.

My idea would be:
* introduce `getParameters` into `HyperwalletPaginationOptions`
** to make this change happen, the date conversion needs to be moved from `Hyperwallet` to `HyperwalletPaginationOptions`
* `HyperwalletUserListOptions` would override `getParameters`, take the parent params and add its own
* `paginate` or `addParameters`in `Hyperwallet` could be removed with one of the others